### PR TITLE
prov/verbs: set of bug-fixes for the verbs provider (SGL + fi_accept())

### PR DESCRIPTION
**Pull request by sydidelot**
Here is a set of two patches for the verbs provider.
Thanks for the review!

### CODE REVIEW

**Comment by sydidelot** [comment]
Does someone know why the "Intel CI" test constantly fails?
Also, would it be possible to make the reason for the failure accessible through Github, like it's done for the other tests?
Thank you!

**Comment by shefty** [comment]
The Intel CI is internal to our group.  We haven't found a way to expose the build failure reason without also exposing internal information about our servers.  And we want to avoid the IT police.  The error for this PR is:

```
fatal: write error: Bad file descriptor
fatal: index-pack failed
```

Other PRs may have a similar issue.  This looks like some problem with the Intel build servers.

**Comment by a-ilango** [comment]
There was an issue with a node that was doing the builds. I've moved the build worker to another node. The tests should run fine now (hopefully).

File: prov/verbs/src/verbs_eq.c [hunk 1]
```diff
@@ -520,6 +520,10 @@ fi_ibv_eq_cm_process_event(struct fi_ibv_eq *eq,
 			return fi_ibv_eq_xrc_connected_event(eq, cma_event,
 							     entry, len, acked);
 		entry->info = NULL;
+		if (priv_datalen) {
+			cm_hdr = priv_data;
+			memcpy(&entry->data, cm_hdr->data, cm_hdr->size);
+		}
```
Comment (a-ilango) [inline prov/verbs/src/verbs_eq.c]: `fi_ibv_eq_copy_event_data` copies the data already. Why do we need this?

Comment (shefty) [reply prov/verbs/src/verbs_eq.c]: Note that fabtests has a cm_data test that verifies the CM connect/accept data exchange.

File: prov/verbs/src/verbs_rma.c [hunk 2]
```diff
@@ -90,6 +90,9 @@ fi_ibv_msg_ep_rma_writemsg(struct fid_ep *ep_fid, const struct fi_msg_rma *msg,
 		.wr.rdma.rkey = (uint32_t)msg->rma_iov->key,
 	};
 
+	if (msg->rma_iov_count > ep->info->tx_attr->rma_iov_limit)
+		return -FI_EINVAL;
```
Comment (shefty) [inline prov/verbs/src/verbs_rma.c]: The rma iov limits are reported as part of the fi_info attributes.  The application is expected to adhere to these limits.  We don't want validation checks in the fast path, as these indicate that the application has been coded incorrectly and is not a run time error.  At best these should be asserts.

File: prov/verbs/src/verbs_eq.c (hunk 1 above)
Comment (sydidelot) [reply prov/verbs/src/verbs_eq.c]: Thanks a lot for your answer. Indeed, it seems that the code already implements CM accept data exchange in `fi_ibv_eq_copy_event_data`. I will do further testing.
For now, I'm closing the PR. 
Sorry for the noise and thanks again for your review.

File: prov/verbs/src/verbs_rma.c (hunk 2 above)
Comment (sydidelot) [reply prov/verbs/src/verbs_rma.c]: Thanks for the comment. I understand your argument and I will remove the checks from the fast path. I'm closing the PR.

**Pull request closed without merging**